fs/fcb/src/fcb.c

```diff
@@ -257,14 +286,21 @@ fcb_offset_last_n(struct fcb *fcb, uint8_t entries,
 int
 fcb_clear(struct fcb *fcb)
 {
-    int rc;
+    int rc = 0;
+
+    rc = os_mutex_pend(&fcb->f_mtx, OS_WAIT_FOREVER);
+    if (rc && rc != OS_NOT_STARTED) {
+        return FCB_ERR_ARGS;
+    }
 
-    rc = 0;
     while (!fcb_is_empty(fcb)) {
```

Comment:
so if there is no fsb_is_empty_nolock() this should handle error checking from  fsb_is_empty()

same for fcb_free_sector_cnt() and other that had mutex added if those are used internally

Reply:
Added `fcb_is_empty_nolock()`